crates/chia-consensus/src/spendbundle_conditions.rs

```diff
@@ -75,6 +76,9 @@ pub fn run_spendbundle(
         subtract_cost(a, &mut cost_left, clvm_cost)?;
 
         let buf = tree_hash(a, puz);
+        if coin_spend.coin.puzzle_hash != buf.into() {
+            return Err(ValidationErr(puz, ErrorCode::WrongPuzzleHash));
```

Comment:
Perhaps we should have a test that explicitly fails this check?

Reply:
added. I refactored the tests some as well, to make them shorter (and easier to read, I think). They are best reviewed one commit at a time